include/multigrid_pclomp/multi_voxel_grid_covariance_omp.h

```diff
@@ -0,0 +1,414 @@
+        std::vector<int> k_indices;
+        int k = kdtree_.radiusSearch (point, radius, k_indices, k_sqr_distances, max_nn);
+
+        // Find leaves corresponding to neighbors
+        k_leaves.reserve (k);
+        for (std::vector<int>::iterator iter = k_indices.begin (); iter != k_indices.end (); iter++)
+        {
+          auto leaf = leaves_.find(leaf_indices_[*iter]);
+          if (leaf == leaves_.end()) {
+            std::cerr << "error : could not find the leaf corresponding to the voxel" << std::endl;
+            std::cin.ignore(1);
+          }
+          k_leaves.push_back (&(leaf->second));
+        }
+        return k;
+      }
+
+      /** \brief Search for all the nearest occupied voxels of the query point in a given radius.
+       * \note Only voxels containing a sufficient number of points are used.
+       * \param[in] cloud the given query point
+       * \param[in] index a valid index in cloud representing a valid (i.e., finite) query point
+       * \param[in] radius the radius of the sphere bounding all of p_q's neighbors
+       * \param[out] k_leaves the resultant leaves of the neighboring points
+       * \param[out] k_sqr_distances the resultant squared distances to the neighboring points
+       * \param[in] max_nn
+       * \return number of neighbors found
+       */
+      inline int
+      radiusSearch (const PointCloud &cloud, int index, double radius,
+                    std::vector<LeafConstPtr> &k_leaves, std::vector<float> &k_sqr_distances,
+                    unsigned int max_nn = 0) const
+      {
+        if (index >= static_cast<int> (cloud.points.size ()) || index < 0)
+          return (0);
+        return (radiusSearch (cloud.points[index], radius, k_leaves, k_sqr_distances, max_nn));
+      }
+
+      PointCloud getVoxelPCD () const
+      {
+        return *voxel_centroids_ptr_;
+      }
+
+  		std::vector<std::string> getCurrentMapIDs() const
+      {
+        std::vector<std::string> output{};
+        for (const auto &element: grid_leaves_) {
+          output.push_back(element.first);
+        }
+        return output;
+      }
+
+    protected:
+
+      /** \brief Filter cloud and initializes voxel structure.
+       * \param[out] output cloud containing centroids of voxels containing a sufficient number of points
+       */
+      void applyFilter (const PointCloudConstPtr &input, const std::string &grid_id, LeafDict &leaves) const;
+
+      void updateVoxelCentroids (const Leaf &leaf, PointCloud &voxel_centroids) const;
+
+      void updateLeaf (const PointT &point, const int &centroid_size, Leaf &leaf) const;
+
+      void computeLeafParams (const Eigen::Vector3d &pt_sum,
+        Eigen::SelfAdjointEigenSolver<Eigen::Matrix3d> &eigensolver,
+        Leaf &leaf) const;
+
+      LeafID getLeafID (const std::string &grid_id, const PointT &point, const BoundingBox &bbox) const;
+
+      /** \brief Flag to determine if voxel structure is searchable. */
+      bool searchable_;
```

Comment:
`searchable_` does not have any rolls (It must always be true. ) so you can remove it completely.

Reply:
Thanks, I removed the variable: https://github.com/tier4/ndt_omp/pull/19/commits/d0f7829ce67968b245b1f13a5745331ec157f542